graphdatascience/graph/graph_result.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass
+from types import TracebackType
+from typing import Optional, Type, TypeVar
+
+from ..graph.graph_object import Graph
+
+TGraphResult = TypeVar("TGraphResult", bound="GraphResult")
+
+
+@dataclass(frozen=True)
+class GraphResult:
+    graph: Graph
+    result: "Series[Any]"
+
+    def __iter__(self):
+        return iter((self.graph, self.result))
+
+    def __enter__(self: TGraphResult) -> Graph:
+        return self.graph
+
+    def __exit__(
+        self,
+        exception_type: Optional[Type[BaseException]],
+        exception_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        self.graph.drop()
+
```

Comment:
I think there should also be `graph` and `result` attributes so that the user doesn't have to use tuple indexing (though it should of course be there for compat)